source/utils/CMakeLists.txt

```diff
@@ -35,6 +35,20 @@ file(GENERATE
   INPUT ${PROJECT_BINARY_DIR}/bpls.cmake.gen
 )
 
+# BPCMP
+if(ADIOS2_HAVE_Python AND Python_Interpreter_FOUND)
+  add_subdirectory(bpcmp)
+
+  configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/bpcmp/bpcmp.cmake.gen.in
+    ${PROJECT_BINARY_DIR}/bpcmp.cmake.gen
```

Comment:
I am doing the same thing as `bpls`. Why would `bpcmp` use a different path? Both are utility scipts.

Reply:
@anagainaru my bad I refered to this: `CMAKE_CURRENT_BINARY_DIR` this variable correctly refers to the the same directory hierarchy in the build directory as in the source dir. I can see that bpls does not do the same thing.

I understand that we this is a conundrum of consistency vs correctness. I do not feel very strong because of the point that you raised. On the other hand is never late to start doing things "Better" having a build dir that correspond to the source dir helps a lot. Ideally I would ask you to also change bpls as you are already changing things but it is unreasonable to do it here, perhaps otherwise a comment saying we are starting to do it this way can help. I will approve your PR in any case that you choose